### PR TITLE
Integrate new DeviceHistoryRPC

### PR DESCRIPTION
**Pull request by cjb**
@keybase/react-hackers 

This PR switches to using the new DeviceHistoryListRPC, which contains extra information about provisioning and revocation.

Previously we were passing the RPC payload directly to our components, so this also specifies a new Device type and uses it in the reducer, and updates the render components and dumb-sheet component's mock data to use it.

No visible UI changes.

### CODE REVIEW

File: shared/devices/index.js [hunk 1]
```diff
@@ -35,7 +35,8 @@ class Devices extends Component {
   render () {
     return (
       <Render
-        devices={this.props.devices}
+        devices={this.props.devices && this.props.devices.filter(dev => !dev.revokedAt)}
+        revokedDevices={this.props.devices && this.props.devices.filter(dev => dev.revokedAt)}
         waitingForServer={this.props.waitingForServer}
```
Comment (cjb) [inline shared/devices/index.js]: The efficiency-purist in me would like a:
```js
{devices, revokedDevices} = this.props.devices.splitOn(dev => dev.revokedAt)
```
func that makes both arrays at once.  :)  Ah!  Found one:

https://lodash.com/docs#partition

Shall I use it?

File: shared/constants/types/more.js [hunk 2]
```diff
@@ -1,10 +1,20 @@
...
-export type Device = {type: DeviceType} & _Device
+import type {Device as _Device, DeviceID, Time} from './flow-types'
+
+export type Device = {
+  name: string;
+  deviceID: DeviceID;
+  type: DeviceType;
+  created: Time;
+  currentDevice: boolean;
+  provisioner: ?_Device;
```
Comment (MarcoPolo) [inline shared/constants/types/more.js]: Do you ever use the Provisioner?

File: shared/devices/index.js (hunk 1 above)
Comment (MarcoPolo) [inline shared/devices/index.js]: 👍

File: shared/constants/types/more.js (hunk 2 above)
Comment (cjb) [inline shared/constants/types/more.js]: Not right now, I haven't looked at the devices/device-page integration yet, maybe it'll go on there?

(The first device has no provisioner btw.)

Comment (chrisnojima) [inline shared/constants/types/more.js]: This is used in the details page for a single device. it's used in the 'timeline' view

File: shared/devices/render.desktop.js [hunk 3]
```diff
@@ -83,8 +84,8 @@ const RevokedDescription = () => (
 
 const RevokedDevices = ({revokedDevices}) => (
   <RevokedHeader>
-    <RevokedDescription />
-    {revokedDevices.map(device => <DeviceRow key={device.name} device={device} revoked />)}
+    <RevokedDescription/>
```
Comment (chrisnojima) [inline shared/devices/render.desktop.js]: i believe those whitespaces are supposed to be in there with the new eslint?

**Comment by chrisnojima** [comment]
👍

**Pull request merged**